packages/hardhat/src/index.ts

```diff
@@ -43,15 +44,19 @@ subtask(TASK_COMPILE_SOLIDITY_COMPILE_JOBS, 'Compiles the entire project, buildi
 
     // RUN TYPECHAIN TASK
     const typechainCfg = config.typechain
-    // incremental generation is only supported in 'ethers-v5'
+    // incremental generation is only supported in 'ethers-v5' and external artifacts
     // @todo: probably targets should specify somehow if then support incremental generation this won't work with custom targets
-    const needsFullRebuild = taskArgsStore.fullRebuild || typechainCfg.target !== 'ethers-v5'
+    const needsFullRebuild =
```

Comment:
Can we somehow support incremental generation with `externalArtifacts`? If I get it correctly we have no way of checking if we already generated types for externalArtifacts and this is the reason for this change. But maybe we can always generate typings only for `externalArtifacts` and avoid regenerating all types for internal files? Incremental generation is often a lifesaver.